source/plot/ShapePlotData2D.ooc

```diff
@@ -0,0 +1,100 @@
+use ooc-collections
+use ooc-math
+use ooc-draw
+import math
+import PlotData2D
+import svg/Shapes
+
+AnyShape: cover {
+	shape: Shape
+	region: FloatBox2D
+	colorBgra: ColorBgra
+	init: func@ (=shape, =region, =colorBgra)
+}
+
+ShapePlotData2D: class extends PlotData2D {
+	_shapes: VectorList<AnyShape>
+
+	init: func (label := "", colorBgra := ColorBgra new()) {
+		super(label, colorBgra)
+		this _shapes = VectorList<AnyShape> new()
+	}
+
+	addRectangle: func (box: FloatBox2D, colorBgra := ColorBgra new()) {
+		this _shapes add(AnyShape new(Shape Rectangle, box, colorBgra))
+	}
+	addCircle: func (center: FloatPoint2D, radius: Float, colorBgra := ColorBgra new()) {
+		this addEllipse(center, radius, radius, colorBgra)
+	}
+	addEllipse: func (center: FloatPoint2D, radiusX, radiusY: Float, colorBgra := ColorBgra new()) {
+		this _shapes add(AnyShape new(Shape Ellipse, FloatBox2D new(center x - radiusX, center y - radiusY, radiusX * 2.0f, radiusY * 2.0f), colorBgra))
+	}
+
```

Comment:
Several empty lines between functions here, we don't usually have that. Please remove.